### PR TITLE
Fixes to launch Magento with PHP 8.1

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Fixes for compatibility with PHP 8.1 (to be able to launch Magento)

### Fixed Issues
1. Fixes magento/magento2#34505

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Analytics/Model/ReportXml/ModuleIterator.php [hunk 1]
```diff
@@ -34,6 +34,7 @@ public function __construct(
      *
      * @return array
      */
+    #[\ReturnTypeWillChange]
     public function current()
```
Comment (fascinosum) [inline app/code/Magento/Analytics/Model/ReportXml/ModuleIterator.php:38]: I would recommend to specify a proper return type for this method instead

File: app/code/Magento/Analytics/ReportXml/Query.php [hunk 2]
```diff
@@ -82,6 +82,7 @@ public function getConfig()
      * @return mixed data which can be serialized by <b>json_encode</b>,
      * which is a value of any type other than a resource.
      */
+    #[\ReturnTypeWillChange]
```
Comment (fascinosum) [inline app/code/Magento/Analytics/ReportXml/Query.php:85]: I would recommend to specify a proper return type for this method instead

File: app/code/Magento/Backend/App/DefaultPath.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ class DefaultPath implements \Magento\Framework\App\DefaultPathInterface
      */
     public function __construct(\Magento\Backend\App\ConfigInterface $config)
     {
-        $pathParts = explode('/', $config->getValue('web/default/admin'));
+        $pathParts = explode('/', $config->getValue('web/default/admin') ?? '');
```
Comment (fascinosum) [inline app/code/Magento/Backend/App/DefaultPath.php]: will `if` statement be more elegant for this case?

File: app/code/Magento/Backend/view/adminhtml/templates/widget/tabshoriz.phtml [hunk 4]
```diff
@@ -19,7 +19,7 @@ use Magento\Framework\View\Helper\SecureHtmlRenderer;
     <?php foreach ($tabs as $_tab): ?>
         <?php $tabId = $block->getTabId($_tab) ?>
         <?php $_tabClass = 'tab-item-link ' . $block->getTabClass($_tab) . ' ' .
-            (preg_match('/\s?ajax\s?/', $_tab->getClass()) ? 'notloaded' : '') ?>
+            (preg_match('/\s?ajax\s?/', $_tab->getClass() ?? '') ? 'notloaded' : '') ?>
```
Comment (fascinosum) [inline app/code/Magento/Backend/view/adminhtml/templates/widget/tabshoriz.phtml:22]: Should it be fixed in `\Magento\Backend\Block\Widget\Tabs::getTabClass` intead?

File: app/code/Magento/Shipping/Model/Simplexml/Element.php [hunk 5]
```diff
@@ -20,6 +20,7 @@ class Element extends \Magento\Framework\Simplexml\Element
      * @param string $namespace If specified, the namespace to which the attribute belongs.
      * @return void
      */
+    #[\ReturnTypeWillChange]
```
Comment (fascinosum) [inline app/code/Magento/Shipping/Model/Simplexml/Element.php]: this is a non-API class, we can add appropriate return types

File: app/code/Magento/Analytics/Model/ReportXml/ModuleIterator.php (hunk 1 above)
Comment (karyna-t) [reply app/code/Magento/Analytics/Model/ReportXml/ModuleIterator.php:38]: Hi @fascinosum , in PHP >= 8 the return type for this method is _mixed_  (https://www.php.net/manual/en/class.iteratoriterator.php) , but in PHP 7.4 this return type is not specified yet.

File: app/code/Magento/Analytics/ReportXml/Query.php (hunk 2 above)
Comment (karyna-t) [reply app/code/Magento/Analytics/ReportXml/Query.php:85]: Hi @fascinosum , in PHP >= 8 the return type for this method is _mixed_  (https://www.php.net/manual/en/jsonserializable.jsonserialize.php) , but in PHP 7.4 this return type is not specified yet.

File: app/code/Magento/Backend/view/adminhtml/templates/widget/tabshoriz.phtml (hunk 4 above)
Comment (karyna-t) [reply app/code/Magento/Backend/view/adminhtml/templates/widget/tabshoriz.phtml:22]: @fascinosum  the problem is in a method 'getClass', and $_tab can be a DataObject

**Comment by karyna-tsymbal-atwix** [comment]
@magento run all tests with env PHP 7.4

**Comment by andrewbess** [comment]
Closed due to this pull request has been divided by separated PRs ([34637](https://github.com/magento/magento2/pull/34637), [34638](https://github.com/magento/magento2/pull/34638), [34640](https://github.com/magento/magento2/pull/34640), [34641](https://github.com/magento/magento2/pull/34641))

**Pull request closed without merging**